apps/rush-lib/src/utilities/VersionControl.ts

```diff
@@ -48,49 +66,50 @@ export class VersionControl {
    * @param repositoryUrl - repository url
    */
   public static getRemoteMasterBranch(repositoryUrl?: string): string {
-    const defaultRemote: string = 'origin';
-    const defaultMaster: string = 'origin/master';
-    let useDefault: boolean = false;
+    const defaultMaster: string = `${DEFAULT_REMOTE}/${DEFAULT_BRANCH}`;
     let matchingRemotes: string[] = [];
 
-    if (!repositoryUrl) {
-      useDefault = true;
-    } else {
+    if (repositoryUrl) {
       const output: string = child_process
-      .execSync(`git remote`)
-      .toString();
+        .execSync(`git remote`)
+        .toString();
       matchingRemotes = output.split('\n').filter(remoteName => {
         if (remoteName) {
           const remoteUrl: string = child_process.execSync(`git remote get-url ${remoteName}`)
             .toString()
             .trim();
-          if (remoteName === defaultRemote && remoteUrl === repositoryUrl) {
-            useDefault = true;
-          }
           return remoteUrl === repositoryUrl;
         }
         return false;
       });
+    } else {
+      console.log(colors.yellow(
+        'A remote URL has not been specified in rush.json. Setting the baseline remote URL is recommended.'
+      ));
+      return defaultMaster;
     }
 
-    if (useDefault) {
-      return defaultMaster;
-    } else if (matchingRemotes.length > 0) {
+    if (matchingRemotes.length > 0) {
       if (matchingRemotes.length > 1) {
-        console.log(`More than one remotes match the repository url. Use the first remote.`);
+        console.log(`More than one remote matches the repository URL. Using the first remote (${matchingRemotes[0]}).`);
       }
-      return `${matchingRemotes[0]}/master`;
+
+      return `${matchingRemotes[0]}/${DEFAULT_BRANCH}`;
+    } else {
+      console.log(colors.yellow(
+        `Unable to find a remote matching the repository URL (${matchingRemotes[0]}). ` +
```

Comment:
We skip `_tryFetchRemoteBranch()` in this case, right?

Reply:
Yeah. That'll be covered by the "don't pull from `origin/master`" case.